### PR TITLE
Divide and conquer algorithms

### PR DESCRIPTION
**Pull request by Dikshit-Kapoor**
#### Checklist

- [ ] Added description of change
- [ ] Added file name matches [File name guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTION.md#New-File-Name-guidelines)
- [ ] Added tests and example, test must pass
- [ ] Relevant documentation/comments is changed or added
- [ ] PR title follows semantic [commit guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTION.md#Commit-Guidelines)
- [ ] Search previous suggestions before making a new one, as yours may be a duplicate.
- [ ] Sort by alphabetical order
- [ ] I acknowledge that all my contributions will be made under the project's license.

Notes:

### CODE REVIEW

**Comment by deadshotsb** [comment]
@Dikshit-Kapoor  Kindly add a description to your changes else it is very difficult to understand what changes you are requesting

**Comment by Dikshit-Kapoor** [comment]
@deadshotsb I have added algorithms based on divide and conquer and their respective pseudo code for better understanding

**Comment by Dikshit-Kapoor** [comment]
A description of  divide and conquer algorithms with a working example

**Comment by kvedala** [comment]
1. code does not compile
2. filenames and folder names should not have spaces

please read the contribution guidelines and make the necessary changes.

File: divide and conquer algorithm/Binary search tree.txt [hunk 1]
```diff
@@ -0,0 +1,129 @@
+A Binary Search Tree (BST) is a tree in which all the nodes follow the below-mentioned properties −
+
```
Comment (kvedala) [inline divide and conquer algorithm/Binary search tree.txt:2]: use comment blocks

File: divide and conquer algorithm/Binary search tree.txt [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+               return;
+            }
+         }
+      }            
+   }
+}        
+
+Example of Binary Serach tree-
+
+#include<stdio.h>
```
Comment (kvedala) [inline divide and conquer algorithm/Binary search tree.txt:61]: these are "C" includes and not "C++" includes
Please use appropriate ones

File: divide and conquer algorithm/Binary search tree.txt [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+    struct node *right;   
+};  
+struct node *root;  
+void main ()  
+{  
+    int choice,item;  
+    do   
+    {  
+        printf("\nEnter the item which you want to insert?\n");  
+        scanf("%d",&item);
```
Comment (kvedala) [inline divide and conquer algorithm/Binary search tree.txt:77]: `printf` and `scanf` are C functions
for C++, use `std::cout` and `std::cin`

File: divide and conquer algorithm/Merge sort.txt [hunk 4]
```diff
@@ -0,0 +1,133 @@
+Merge sort-Merge sort keeps on dividing the list into equal halves until it can no more be divided. By definition, if it is only one element in the list, it is sorted.
```
Comment (kvedala) [inline divide and conquer algorithm/Merge sort.txt:1]: there is already a "sorting" folder. No need for a new folder.

Comment (kvedala) [inline divide and conquer algorithm/Merge sort.txt:1]: TXT file - 
please describe its application for the pull-request

File: divide and conquer algorithm/Quick sort.txt [hunk 5]
```diff
@@ -0,0 +1,99 @@
+Quick sort-Quick sort is a highly efficient sorting algorithm and is based on partitioning of array of data into smaller arrays. 
+           A large array is partitioned into two arrays one of which holds values smaller than the specified value, say pivot, based on
+           which the partition is made and another array holds values greater than the pivot value.
```
Comment (kvedala) [inline divide and conquer algorithm/Quick sort.txt:3]: check with existing implementations in the "sorting" folder

File: divide and conquer algorithm/Strassens multiplication.txt [hunk 6]
```diff
@@ -0,0 +1,89 @@
+Naive method-
+Algorithm: Matrix-Multiplication (X, Y, Z) 
+for i = 1 to p do
```
Comment (kvedala) [inline divide and conquer algorithm/Strassens multiplication.txt:3]: are these comments or code?
if comments, why are they not in comment blocks?

File: divide and conquer algorithm/Strassens multiplication.txt [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+
+I:=M2+M3−M6−M7
+J:=M4+M6
+K:=M5+M7
+L:=M1−M3−M4−M5
+
+
+Example of Strassen's Multiplication-
+
+#include <stdio.h>
```
Comment (kvedala) [inline divide and conquer algorithm/Strassens multiplication.txt:33]: is this a C program or a C++ program?

**Comment by kvedala** [comment]
@Dikshit-Kapoor are you still interested in contributing to the repo with the above requested fixes?

File: divide and conquer algorithm/Binary search tree.txt (hunk 2 above)
Comment (Panquesito7) [reply divide and conquer algorithm/Binary search tree.txt:61]: Use `cstdio` and `cstdlib`.

File: divide and conquer algorithm/Binary search tree.txt [hunk 8]
```diff
@@ -0,0 +1,129 @@
...
+#include<stdlib.h>  
+void insert(int);  
+struct node  
+{  
+    int data;  
+    struct node *left;   
+    struct node *right;   
+};  
+struct node *root;  
+void main ()
```
Comment (Panquesito7) [inline divide and conquer algorithm/Binary search tree.txt:71]: Preferably, use `int` and avoid using `void`.

File: divide and conquer algorithm/Strassens multiplication.txt [hunk 9]
```diff
@@ -0,0 +1,89 @@
...
+	for(i=0;i<2;i++)
+	{
+		printf("\n");
+		for(j=0;j<2;j++)
+			printf("%d\t",c[i][j]);
+	}
+
+	return 0;
+}
+
```
Comment (Panquesito7) [inline divide and conquer algorithm/Strassens multiplication.txt:89]: Remove this empty line.

**Pull request closed without merging**